src/pseudopeople/loader.py

```diff
@@ -9,19 +9,40 @@
 from pseudopeople.exceptions import DataSourceError
 
 
-def load_standard_dataset_file(data_path: Path, user_filters: List[Tuple]) -> pd.DataFrame:
-    if data_path.suffix == ".parquet":
+def load_data(
+    input_data: Union[Path, pd.DataFrame], user_filters: List[Tuple]
+) -> pd.DataFrame:
+    if isinstance(input_data, pd.DataFrame):
+        data = filter_data(input_data, user_filters)
+    else:
+        data = load_standard_dataset_file(input_data, user_filters)
+
+    return data
+
+
+def filter_data(input_data: pd.DataFrame, user_filters: List[Tuple]) -> pd.DataFrame:
+    if len(user_filters) > 0:
+        for filter in user_filters:
+            input_data = input_data.query(f"{filter[0]} {filter[1]} '{filter[2]}'")
```

Comment:
Also, I have no clue what this represents. Maybe use a destructuring like `for foo, bar, baz in user_filters:` to give these names?